check_diff/src/lib.rs

```diff
@@ -56,3 +136,139 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
+    let Ok(command) = Command::new("rustc").args(["--print", "sysroot"]).output() else {
+        return Err(CheckDiffError::FailedCommand(
+            "Error getting sysroot".to_string(),
+        ));
+    };
+
+    let Ok(sysroot) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting sysroot to string".to_string(),
+        ));
+    };
+
+    let ld_lib_path = format!("{}/lib", sysroot.trim_end());
+    return Ok(ld_lib_path);
+}
+
+pub fn get_cargo_version() -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new("cargo").args(["--version"]).output() else {
+        return Err(CheckDiffError::FailedCargoVersion(
+            "Failed to obtain cargo version".to_string(),
+        ));
+    };
+
+    let Ok(cargo_version) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting cargo version to string".to_string(),
+        ));
+    };
+
+    return Ok(cargo_version);
+}
+
+pub fn copy_src_to_dst(src: &str, dst: &str) -> Result<(), CheckDiffError> {
+    let Ok(_) = Command::new("cp").args([src, dst]).output() else {
+        return Err(CheckDiffError::FailedCopy(
+            "Error copying rustfmt release to destination".to_string(),
+        ));
+    };
+    return Ok(());
+}
+
+pub fn build_rsfmt_from_src(ld_lib_path: &String) -> Result<(), CheckDiffError> {
+    let Ok(_) = Command::new("cargo")
+        .env("LD_LIB_PATH", ld_lib_path)
+        .args(["build", "-q", "--release", "--bin", "rustfmt"])
+        .output()
+    else {
+        return Err(CheckDiffError::FailedSourceBuild(
+            "Error building rustfmt from source".to_string(),
+        ));
+    };
+    return Ok(());
+}
+
+pub fn get_binary_version(binary: String) -> Result<String, CheckDiffError> {
+    let Ok(command) = Command::new(binary.as_str()).args(["--version"]).output() else {
+        return Err(CheckDiffError::FailedVersioning(format!(
+            "Failed to get version for {}",
+            binary
+        )));
+    };
+
+    let Ok(binary_version) = String::from_utf8(command.stdout) else {
+        return Err(CheckDiffError::FailedUtf8(
+            "Error converting binary version to string".to_string(),
+        ));
+    };
+    return Ok(binary_version);
+}
+
+// Compiles and produces two rustfmt binaries.
+// One for the current master, and another for the feature branch
+// Parameters:
+// dest: Directory where rustfmt will be cloned
+pub fn compile_rustfmt(
+    dest: &Path,
+    remote_repo_url: String,
+    feature_branch: String,
+    commit_hash: Option<String>,
+) -> Result<Command, CheckDiffError> {
+    const RUSTFMT_REPO: &str = "https://github.com/rust-lang/rustfmt.git";
+
+    let _clone_git_result = clone_git_repo(RUSTFMT_REPO, dest)?;
+    let _git_remote_add_result = git_remote_add(remote_repo_url.as_str())?;
+    let _git_fetch_result = git_fetch(feature_branch.as_str())?;
```

Comment:
Does the compiler complain here? In the `Ok` case each of these functions returns `()`, so if we don't need to then I'd say let's not assign anything to these variables.

Reply:
similar comment applies to the other `_*` variables below.

Reply:
yep the compiler requires assignment of variables. it accepts '_' as a variable name, however. ill change it accordingly

Reply:
I'm fairly certain that with the `?` operator the compiler won't complain.

<img width="451" alt="Screen Shot 2024-08-19 at 9 09 49 PM" src="https://github.com/user-attachments/assets/8bee9baa-e3ec-4c7c-bdb0-64eeac3a2bfd">

Reply:
ah maybe when i dint prepend `_` is when the compiler complaint. Otherwise, i totally agree that not having variables binded to these functions makes the most sense too. ill change it accordingly